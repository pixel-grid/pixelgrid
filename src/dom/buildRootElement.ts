import { BuildLayoutOptions } from '../grid.interfaces';

/**
 * Create empty div element and append it to the end of the body
 * @param id Element's identifier
 */
const buildRootElement: (id: string, options: BuildLayoutOptions) => HTMLElement = (id, options) => {
    const result = document.createElement('div');

    result.id = id;
    result.style.position = 'absolute';
    result.style.top = '0';
    result.style.left = '0';
    result.style.right = '0';
    result.style.pointerEvents = 'none';
    result.style.overflow = 'hidden';

    if (options.zIndex !== undefined) {
        result.style.zIndex = `${options.zIndex}`;
    }

    document.body.appendChild(result);

    return result;
};

export default buildRootElement;
