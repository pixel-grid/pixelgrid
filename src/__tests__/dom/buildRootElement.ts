import buildRootElement from '../../dom/buildRootElement';

describe('buildRootElement', () => {
    test('builds empty root element', () => {
        const result = buildRootElement('pixelgrid', {});
        expect(result).toMatchSnapshot();
    });

    test('builds root element with custom zIndex', () => {
        const result = buildRootElement('pixelgrid', { zIndex: 123456 });
        expect(result).toMatchSnapshot();
    });
});
