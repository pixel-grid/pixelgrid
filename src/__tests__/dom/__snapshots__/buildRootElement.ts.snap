// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildRootElement builds empty root element 1`] = `
<div
  id="pixelgrid"
  style="position: absolute; top: 0px; left: 0px; right: 0px; pointer-events: none; overflow: hidden;"
/>
`;

exports[`buildRootElement builds root element with custom zIndex 1`] = `
<div
  id="pixelgrid"
  style="position: absolute; top: 0px; left: 0px; right: 0px; pointer-events: none; overflow: hidden; z-index: 123456;"
/>
`;
